SofaKernel/framework/sofa/helper/system/thread/CircularQueue.inl

```diff
@@ -271,14 +271,17 @@ inline SOFA_HELPER_API bool ManyThreadsPerEnd::pop(AtomicInt array[], int maxSiz
         return false;
     }
     // atomically reserve the element to read
-    int readIdx = head.exchange_and_add(1) & (maxCapacity-1); // maxCapacity is assumed to be a power of 2
+    int readIdx = head.fetch_add(1) & (maxCapacity-1); // maxCapacity is assumed to be a power of 2
 
     // Active wait:
     // loop as long as other threads have not put any valid value in the element.
     // It happens when the queue is temporarily empty.
-    while((item = array[readIdx]) == -1)
+    do
     {
+        item.exchange(array[readIdx]);
     }
+    while(item == -1);
+    //while((item = array[readIdx]) == -1)
```

Comment:
Keeping the old line on purpose?

Reply:
Just a quick remark:
In general  I always prefer to explicitly call the load/store functions for read/write atomic operations like this:

do
{
     item.exchange(array[readIdx].load() );
} while(item.load() == -1);

It helps to quickly understand or remind the developers that it's an atomic operation and it allows to set the memory order model to do some optimization. 
But in this case the memory order is not specified (default value) so explicitly call the load/store function call is only to make clear that there is an atomic read/write operation.
I see that explicitly call the load/store functions was not previously used so I don't know if it worths to check all the atomic operations.